dsperse/src/slice/onnx_slicer.py

```
@@ -261,276 +377,345 @@ def _get_segment_details(segment_nodes, graph, initializer_map):
-                try:
-                    model_with_shapes = shape_inference.infer_shapes(model)
-                    model = model_with_shapes
-                    logger.info(f"Shape inference successful for {path}")
-                    print(f"Shape inference successful for {path}")
-                except Exception as shape_error:
-                    logger.warning(f"Shape inference failed for {path}: {shape_error}")
-                    print(f"Shape inference failed for {path}: {shape_error}")
-                    # Continue with original model if shape inference fails
-
-                # Concretize any remaining symbolic dims to batch=1 for ezkl
+    def _process_single_slice(path: str, traced_shapes: dict = None) -> str | None:
```

Ruff (0.14.13)

PEP 484 prohibits implicit `Optional`

Convert to `T | None`

(RUF013)